lib: posix: Move posix layer from 'kernel' to 'lib'

ramakrishnapallala opened the pull request:
1. Moved posix sources from kernel to lib folder
2. Fixed posix tests header file dependencies
3. Fixed Coverity Issues #6677 #6694

tyrantlizard77 reviewed (approved):
code looks fine, but please address CI failures, seems like Nios II is failing a lot of tests

ramakrishnapallala commented:
Yes @andrewboie I am working on resolving the CI issues.

Commit:
lib: posix: Move posix layer from 'kernel' to 'lib'

Move posix layer from 'kernel' to 'lib' folder as it is not
a core kernel feature.

Fixed posix header file dependencies as part of the move and
also removed NEWLIBC related macros from posix headers.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

Commit:
tests: posix: Resolve header file dependencies

Fixed header file dependencies by reordering the
inclusion sequence in posix tests.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

Commit:
lib: posix: Fix mutex locking in pthread_cancel

Fix mutex locking sequence in pthread_cancel()

Coverity-CID: 183055

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

Commit:
lib: posix: Fix integer overflow in timer_gettime

Fix 'Unintentional integer overflow' coverity issue
in timer_gettime().

Coverity-CID: 183038

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

ramakrishnapallala commented:
@nashif rebased the PR to the latest and also pushed the Coverity fixes. Please have a look.
